### PR TITLE
Pull request for qa notebook

### PR DESCRIPTION
**Pull request by cecileloge**

### CODE REVIEW

**Comment by cecileloge** [comment]
@googlebot I fixed it.

**Comment by cecileloge** [comment]
@googlebot I fixed it.

File: courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb [hunk 1]
```diff
@@ -0,0 +1,684 @@
+{
```
Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <p>Great intro! Please put the "Learning Objectives" at the top, just below the title.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <p>Use the <a href="https://www.python.org/dev/peps/pep-0008/#imports" target="_blank">pep8</a> convention to format the imports</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <p>* "will input" --&gt; "will take as input"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <ul><li>Break lines according to <a href="https://www.python.org/dev/peps/pep-0008/#maximum-line-length" target="_blank">pep8</a> (no more than 80 chars per line), or, in this case import "pad_sequence" already in the import section</li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <ul><li>"batchSize" --&gt; "batch_size"</li><li>maybe use more informative variable names for "seq_pars" and "seq_ques" as for instance "max_paragraph_length" , etc.</li><li>pep8 : spaces missing after commas in data[['question','context','text','start_word', 'end_word']]</li><li>avoid abbreviations in variable names (it makes the code harder to read). For instance: "ye" --&gt; "y_end_word". Also it seems that the naming you use for "ye" and "ys" should be reversed ("ye" should correspond to "y.end_word" and not "y_start_word". Make sure this doesn't result in a bug later on.</li><li>The name "y" here seems not informative enough. Usually, it's understood as labels - is that the case here?</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <ul><li>pep8: no more than 80 chars per line. </li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <ul><li>pep8: Two blank lines between functions.</li><li>"F1_score" --&gt; "f1_score"</li><li>"F1" --&gt; "f1"</li><li>In general, I'd prefer to use already implemented functions to compute the metrics, as in sklearn.metric</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <ul><li>pep8</li><li>"BaseModel" --&gt; "base_model" (I'd choose either simply "model" or something more informative for that variable name)</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <ul><li>pep8: spaces surrounding "=" in the function arguments</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <ul><li>pep8: (indentation is off by one space, and no space surrounding "=" in function arguments)</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <p>pep8</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <ul><li>variable names: EM, F1 (no upper case)</li><li>pep8</li><li>Is there a way to break that function into smaller functions and give them informative names to understand what happens?</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <ul><li>pep8: too long lines (import "pad_sequence" at the beginning)</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <ul><li>(EM_base,F1_base) -&gt; (em_base, f1_base)</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <ul><li>avoid using a single char for variable names (it's hard to search &amp; replace and not very informative). It's better to use the word itself rather than its abbreviation</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (BenoitDherin) [inline courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <ul><li>pep8</li><li>Maybe add a cell where you use this function.</li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1325/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (cecileloge) [reply courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <p>Thanks for flagging the start/end bug, I had mixed them up when cleaning my code in the last steps... Fixed now!</p>

Comment (cecileloge) [reply courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <p>Because we're doing a F1 Score on string data, this one unfortunately does not exist as is in sklearn!</p>

Comment (cecileloge) [reply courses/machine_learning/deepdive2/supplemental/solutions/question_answering_rnn.ipynb:1]: <p>Broke it into two functions, let me know if better!</p>
